f5_openstack_agent/lbaasv2/drivers/bigip/tenants.py

```diff
@@ -75,8 +75,57 @@ def assure_tenant_created(self, service):
                         "Folder creation error for tenant %s" %
                         (tenant_id))
 
+        if self.conf.use_namespaces and sync:
+            # pzhang sync all route domain on an active bigip to all
+            # standby bigip
+            bigips = self.driver.get_all_bigips()
+            active_bigips = [
+                bigip for bigip in bigips
+                if bigip.failover_state == "active"
+            ]
+
+            standby_bigips = [
+                bigip for bigip in bigips
+                if bigip.failover_state == "standby"
+            ]
+
+            if active_bigips:
+                # pzhang; default to use first
+                # active bigip to sync route domain
+                # of certain parition of the loadbalancer in service.
+                active_bigip = active_bigips[0]
+                rd_ids = self.network_helper.get_route_domain_ids(
+                    active_bigip, partition=folder_name)
+            else:
+                LOG.warning(
+                    "Not found active bigip to sync route domain in L2 mode"
+                )
+
+            if rd_ids:
+                for rd_id in rd_ids:
+                    for standby_bigip in standby_bigips:
+                        rd_exists = self.network_helper.route_domain_exists(
+                            standby_bigip,
+                            folder_name
+                        )
+                        if not rd_exists:
+                            self.network_helper.create_route_domain(
```

Comment:
I am not quite sure about this. But when we tried to create a new route domain in some standby bigips, do we need to store the route domain id in the route domain cache?

Reply:
Why do we need to store route domain id in the route domain cache?

Reply:
> I am not quite sure about this. But when we tried to create a new route domain in some standby bigips, do we need to store the route domain id in the route domain cache?

Why do we need to store route domain id in the route domain cache?

Reply:
When agent tries to create a new route domain in bigip, agent will try to get the next route domain id. I am not sure if get the new route domain id will search the route domain cache or not.  If yes, we need to keep the route domain sync up with the system? 

I am not quite sure about this, just for your information.

Reply:
sync data is not related to creating a new tenant route domain. synced route domains already exist on active bigip, there is no need to get the next domain ids. When previous VIP and members syncing, it will get the route domain from the route domain cache which has fixed vlan or vxlan corresponding to it. such as

```
{
	u '57e89acdfb6e40a2bc7f6185645dbbdd'(这个是 tenant id): {
		1: {
			'vxlan-64': {
				'subnets': {
					u '4927370c-48de-47f2-aea2-b555302a4326': {
						'cidr': IPNetwork('7.7.7.0/24')
					}
				}
			},
			'vxlan-74': {
				'subnets': {
					u 'aa5509e6-8795-461f-91d4-0634025aab30': {
						'cidr': IPNetwork('6.6.6.0/24')
					}
				}
			}
		}
	}
}
```

Reply:
> When agent tries to create a new route domain in bigip, agent will try to get the next route domain id. I am not sure if get the new route domain id will search the route domain cache or not. If yes, we need to keep the route domain sync up with the system?
> 
> I am not quite sure about this, just for your information.

sync data is not related to creating a new tenant route domain. synced route domains already exist on active bigip, there is no need to get the next domain ids. When previous VIP and members are syncing, it will get the route domain from the route domain cache which has fixed vlan or vxlan corresponding to it. such as

```
{
	u '57e89acdfb6e40a2bc7f6185645dbbdd'(这个是 tenant id): {
		1: {
			'vxlan-64': {
				'subnets': {
					u '4927370c-48de-47f2-aea2-b555302a4326': {
						'cidr': IPNetwork('7.7.7.0/24')
					}
				}
			},
			'vxlan-74': {
				'subnets': {
					u 'aa5509e6-8795-461f-91d4-0634025aab30': {
						'cidr': IPNetwork('6.6.6.0/24')
					}
				}
			}
		}
	}
}
```
It will not get route domain ids from the next route domain function.